modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -975,68 +979,114 @@ public abstract class Base${schemaName}ResourceImpl
 		}
 	}
 
-	private Collection<Permission> _getPermissions(long companyId, List<ResourceAction> resourceActions, long resourceId, String resourceName, String[] roleNames) throws Exception {
-		_checkResources(companyId, resourceId, resourceName);
+	private Collection<Permission> _getPermissions(
+		long companyId, List<ResourceAction> resourceActions,
+			long resourceId, String resourceName, String[] roleNames)
+			throws Exception {
 
-		Map<String, Permission> permissions = new LinkedHashMap<>();
+			_checkResources(companyId, resourceId, resourceName);
 
-		List<ResourcePermission> resourcePermissions = new ArrayList<>();
+			Map<String, Permission> permissions = new LinkedHashMap<>();
 
-		try {
-			Method method = ResourcePermissionLocalService.class.getMethod("getResourcePermissions", String.class);
+			List<String> actionIds = new ArrayList<>();
 
-			resourcePermissions = (List<ResourcePermission>)method.invoke(resourcePermissionLocalService, resourceName);
-		}
-		catch (NoSuchMethodException noSuchMethodException) {
-			resourcePermissions = resourcePermissionLocalService.getResourcePermissions(companyId, resourceName, ResourceConstants.SCOPE_INDIVIDUAL, String.valueOf(resourceId));
-		}
+			Set<ResourcePermission> resourcePermissionSet;
+
+			for (ResourceAction resourceAction : resourceActions) {
+				actionIds.add(resourceAction.getActionId());
+			}
+
+			resourcePermissionSet = _getResourcePermissionSet(companyId, resourceId, resourceName);
+
+			for (ResourcePermission resourcePermission : resourcePermissionSet) {
```

Comment:
Refactor this loop as discussed. Following the same logic as edit_permissions.jsp.